### PR TITLE
Add PackagerV3 object

### PR DESCRIPTION
**Pull request by sgallagher**
This should be the complete PackagerV3 object and parser (no emitter). It's admittedly light on tests, but I want to get the review out. We can add to the tests later, assuming this passes muster.

Also included is a patch to handle the new `default` property in the ModulemdProfile object.

### CODE REVIEW

**Comment by sgallagher** [comment]
Looks like there are a couple memory leaks. I'll fix those and push an update.

**Comment by sgallagher** [comment]
I just pushed an update for the specfile test as well, making it a little more useful.

File: modulemd/tests/test-modulemd-packager-v3.c [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    ==,
+    modulemd_packager_v3_get_description (packager));
+
+  strv = modulemd_packager_v3_get_module_licenses_as_strv (packager);
+  g_assert_nonnull (strv);
+  g_assert_nonnull (strv[0]);
+  g_assert_cmpstr ("MIT", ==, strv[0]);
+  g_assert_null (strv[1]);
+  g_clear_pointer (&strv, g_strfreev);
+
```
Comment (mmathesius) [inline modulemd/tests/test-modulemd-packager-v3.c:154]: Verify `xmd`?

File: modulemd/tests/test-modulemd-packager-v3.c [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+  strv = modulemd_packager_v3_get_build_config_contexts_as_strv (packager);
+  g_assert_nonnull (strv);
+  g_assert_nonnull (strv[0]);
+  g_assert_cmpstr ("CTX1", ==, strv[0]);
+  g_assert_nonnull (strv[1]);
+  g_assert_cmpstr ("CTX2", ==, strv[1]);
+  g_assert_null (strv[2]);
+  g_clear_pointer (&strv, g_strfreev);
+
```
Comment (mmathesius) [inline modulemd/tests/test-modulemd-packager-v3.c:163]: Verify `community`, `documentation`, and `tracker`?

**Review by mmathesius — CHANGES_REQUESTED** [review]
Everything looks functional except for some attributes that aren't getting copied by `modulemd_packager_v3_copy()`--although I'm not sure there's an immediate use for that call. So, all of my comments could be future improvements if you don't have time to fix them promptly. Thanks!

File: modulemd/include/private/modulemd-packager-v3.h [hunk 3]
```diff
@@ -0,0 +1,689 @@
...
+ * Returns: (transfer full): A newly-allocated deep-copy of this
+ * #ModulemdPackagerV3 object. This object must be freed with g_object_unref().
+ *
+ * Since: 2.10
+ */
+ModulemdPackagerV3 *
+modulemd_packager_v3_copy (ModulemdPackagerV3 *self);
+
+
+void
```
[the remark this answers is not in the document]
Comment (sgallagher) [reply modulemd/include/private/modulemd-packager-v3.h:92]: Added

File: modulemd/tests/test-modulemd-packager-v3.c (hunk 1 above)
Comment (sgallagher) [reply modulemd/tests/test-modulemd-packager-v3.c:154]: I meant to leave a comment there indicating that verifying XMD would be much easier in a Python test.

File: modulemd/include/private/modulemd-packager-v3.h [hunk 4]
```diff
@@ -0,0 +1,689 @@
...
+ * @self: (in): This #ModulemdPackagerV3 object.
+ * @license: (in): A license to remove from the list. Has no effect if the
+ * license is not present.
+ *
+ * Since: 2.10
+ */
+void
+modulemd_packager_v3_remove_module_license (ModulemdPackagerV3 *self,
+                                            const gchar *license);
+
```
[the remark this answers is not in the document]
Comment (sgallagher) [reply modulemd/include/private/modulemd-packager-v3.h:185]: Yes

**Comment by sgallagher** [comment]
I think I've resolved all of your concerns with this latest push. Thanks for the detailed review!

**Review by mmathesius — APPROVED** [review]

**Comment by mmathesius** [comment]
LGTM. Thank you!

**Pull request merged**